### DISCUSSION TITLE
Selecting the size of clique to return given a specific node

### DISCUSSION THREAD

**Discussion by MDrance** [networkx/networkx#6767] [Q&A] [answered]
Hello,

I was wondering if, given a specific node, I could use Networkx to find all cliques of a given size this node belongs in.
The ```cliques_containing_node()``` function is deprecated and if I'm understanding everything correctly, the ```find_cliques()``` function return only the maximal clique, even if you don't specify the  ```nodes``` parameter.

For example, for a node of interest ```n```, I would like to find all the cliques of size 4 ```n``` belongs to and for each of these cliques, the 3 nodes forming the clique with ```n```.
Any suggestion ?

Best,
Martin Drancé

**Comment by dschult** [comment, marked as answer]
I would suggest finding all the maximal cliques (that is, the cliques with no more nodes possible to add and have it be a clique). For each maximal clique, see if it is large enough to have 4 nodes. If it is, remove `n` from the list of nodes and use something like `itertools.combinations` on the other nodes to get all subsets of size 3. Write the function as a generator yielding the sets of nodes -- you might get a large output set!! :}

**Reply by MDrance** [reply]
Thanks a lot @dschult for you answer, I'll try that and hope it's doable on large graphs ! :)
